controllers/controllers/workloads/cfspace_controller.go

```diff
@@ -351,39 +360,27 @@ func (r *CFSpaceReconciler) finalizeCFApp(ctx context.Context, log logr.Logger,
 	}
 
 	for i := range appList.Items {
-		err = r.client.Delete(ctx, &appList.Items[i])
-		if err != nil {
-			log.Error(err, "failed to delete cfApp", "AppName", appList.Items[i].Name)
-			return err
-		}
-	}
-
-	waitForCFAppDeletion := func(ctx context.Context, log logr.Logger, namespace string, done chan bool) {
-		for {
-			select {
-			case <-time.After(10 * time.Second):
-				log.Info("timeout waiting for CFApps to be deleted")
-				return
-			default:
-			}
-
-			var cfAppList korifiv1alpha1.CFAppList
-			err = r.client.List(ctx, &cfAppList, client.InNamespace(namespace))
+		if appList.Items[i].GetDeletionTimestamp().IsZero() {
+			log.Info(fmt.Sprintf("deleting CFApp %s", appList.Items[i].Name))
+			err = r.client.Delete(ctx, &appList.Items[i], client.PropagationPolicy(metav1.DeletePropagationForeground))
 			if err != nil {
-				log.Error(err, "failed to list CFApps while watching CFSpace")
-				return
+				log.Error(err, "failed to delete CFApp", "AppName", appList.Items[i].Name)
 			}
-
-			if len(cfAppList.Items) == 0 {
-				return
-			}
-
-			time.Sleep(time.Millisecond * 100)
 		}
 	}
 
-	done := make(chan bool, 1)
-	go waitForCFAppDeletion(ctx, log, namespace, done)
+	var cfAppList korifiv1alpha1.CFAppList
+	err = r.client.List(ctx, &cfAppList, client.InNamespace(namespace))
+	if err != nil {
+		log.Error(err, "failed to list CFApps while watching CFSpace")
+		return fmt.Errorf("failed to list CFApps while watching CFSpace")
+	}
+
+	if len(cfAppList.Items) > 0 {
+		err = fmt.Errorf("%d CFApps still found", len(cfAppList.Items))
+		log.Error(err, fmt.Sprintf("%d CFApps still found", len(cfAppList.Items)))
```

Comment:
This should be a log.Info, since I think this case will occur when there isn't any real error

Reply:
Good point. Fixed